lib/StripeObject.php

```diff
@@ -13,45 +13,54 @@
 class StripeObject implements ArrayAccess, JsonSerializable
 {
     /**
-     * @var Util\Set Attributes that should not be sent to the API because
+     * @return Util\Set Attributes that should not be sent to the API because
      *    they're not updatable (e.g. API key, ID).
      */
-    public static $permanentAttributes;
+    public static function getPermanentAttributes()
+    {
+        static $permanentAttributes = null;
+        if ($permanentAttributes === null) {
+            $permanentAttributes = new Util\Set(array('_opts', 'id'));
+        }
+        return $permanentAttributes;
+    }
+
     /**
-     * @var Util\Set Attributes that are nested but still updatable from
+     * @return Util\Set Attributes that are nested but still updatable from
      *    the parent class's URL (e.g. metadata).
      */
-    public static $nestedUpdatableAttributes;
-
-    public static function init()
+    public static function getNestedUpdatableAttributes()
     {
-        self::$permanentAttributes = new Util\Set(array('_opts', 'id'));
-        self::$nestedUpdatableAttributes = new Util\Set(array(
-            // Numbers are in place for indexes in an `additional_owners` array.
-            //
-            // There's a maximum allowed additional owners of 3, but leave the
-            // 4th so errors work properly.
-            0, 1, 2, 3, 4,
-
-            'additional_owners',
-            'address',
-            'address_kana',
-            'address_kanji',
-            'card',
-            'dob',
-            'inventory',
-            'legal_entity',
-            'metadata',
-            'owner',
-            'payout_schedule',
-            'personal_address',
-            'personal_address_kana',
-            'personal_address_kanji',
-            'shipping',
-            'tos_acceptance',
-            'transfer_schedule',
-            'verification',
-        ));
+        static $nestedUpdatableAttributes = null;
+        if ($nestedUpdatableAttributes === null) {
+            $nestedUpdatableAttributes = new Util\Set(array(
+                // Numbers are in place for indexes in an `additional_owners` array.
+                //
+                // There's a maximum allowed additional owners of 3, but leave the
+                // 4th so errors work properly.
+                0, 1, 2, 3, 4,
+
+                'additional_owners',
+                'address',
+                'address_kana',
+                'address_kanji',
+                //'card',
+                'dob',
+                'inventory',
+                'legal_entity',
+                'metadata',
+                'owner',
+                'payout_schedule',
+                'personal_address',
```

Comment:
This lives under `legal_entity` on the `Account` object and `legal_entity` is not a real class/object in the SDK (nor the API). This means you can't define a custom `getNestedUpdatableAttributes` and make it work for some of those params. Would the plan be to "leave" it on `StripeObject` directly? If so, won't most people pushing PR just default to putting it here anyway? Seems hard to realize there are edge-cases when you do.

Reply:
The attributes defined here don't have to be top-level attributes in a given object, they can themselves be nested under other attributes (else the current code would not work either). So my intent was to move all of the identity verification attributes under `Account`, but as said above we can also keep the attributes in `StripeObject` by default and only override the global whitelist when needed.

Reply:
I'd be bullish in moving these under something like `Account` now that class-specific rules are supported. It's always seemed pretty janky to me that things like `0, 1, 2, 3, 4` and `additional_owners` sit at the top level.